### PR TITLE
improve the `histogram_matching` function

### PR DESCRIPTION
**Pull request by huuquan1994**
#### Changes
Update to the new `interp` function so that the `histogram_matching` works properly.

#### Type of change

- [x] 🐞 Bug fix (non-breaking change which fixes an issue)

#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

#### See the visual result here
Comparison based on skimage's histogram matching
https://scikit-image.org/docs/0.16.x/auto_examples/color_exposure/plot_histogram_matching.html
image

### CODE REVIEW

File: kornia/contrib/histogram_matching.py [hunk 1]
```diff
@@ -45,25 +45,23 @@ def histogram_matching(source: torch.Tensor, template: torch.Tensor) -> torch.Te
...
 def interp(x: torch.Tensor, xp: torch.Tensor, fp: torch.Tensor) -> torch.Tensor:
-    """Interpolate ``x`` tensor according to ``xp`` and ``fp`` as in ``np.interp``.
+    """One-dimensional linear interpolation for monotonically increasing sample points.
 
-    This implementation cannot reproduce numpy results identically, but reasonable.
-    Code referred to `here <https://github.com/pytorch/pytorch/issues/1552#issuecomment-926972915>`_.
+    Returns the one-dimensional piecewise linear interpolant to a function with
+    given discrete data points :math:`(xp, fp)`, evaluated at :math:`x`.
+
+    This is confirmed to be a correct implementation.
```
Comment (edgarriba) [inline kornia/contrib/histogram_matching.py:53]: Can you add this two lines as a code comment?

File: kornia/contrib/histogram_matching.py [hunk 2]
```diff
@@ -45,25 +45,23 @@ def histogram_matching(source: torch.Tensor, template: torch.Tensor) -> torch.Te
...
+    See https://github.com/pytorch/pytorch/issues/1552#issuecomment-979998307
 
     Args:
-        x: the input tensor that needs to be interpolated.
-        xp: the x-coordinates of the referred data points.
-        fp: the y-coordinates of the referred data points, same length as ``xp``.
+        x: the :math:`x`-coordinates at which to evaluate the interpolated
```
Comment (edgarriba) [inline kornia/contrib/histogram_matching.py:57]: Add the possible shapes

File: kornia/contrib/histogram_matching.py [hunk 3]
```diff
@@ -45,25 +45,23 @@ def histogram_matching(source: torch.Tensor, template: torch.Tensor) -> torch.Te
...
+            values.
+        xp: the :math:`x`-coordinates of the data points, must be increasing.
+        fp: the :math:`y`-coordinates of the data points, same length as `xp`.
 
     Returns:
-        The interpolated values, same shape as ``x``.
+        the interpolated values, same size as `x`.
```
Comment (edgarriba) [inline kornia/contrib/histogram_matching.py:63]: Include a small snippet example (possibly something illustrative from your example, or tests)

**Review by edgarriba — COMMENTED** [review]
Thanks for the fix. Can we double check the tests so we ensure that we do (and keep) the right thing ? If you have a colab with the example, would be nice to send a separate pr in 
https://github.com/kornia/tutorials

**Comment by huuquan1994** [comment]
@edgarriba sure,

I add a notebook to https://github.com/kornia/tutorials. See the PR here: https://github.com/kornia/tutorials/pull/26

**Comment by edgarriba** [comment]
Awesome, let fix the issues and we can merge

**Comment by huuquan1994** [comment]
> Awesome, let fix the issues and we can merge

Thanks! Feel free to ask if you need anything!

**Comment by edgarriba** [comment]
well most of the tests are still not passing in this PR. I suggest to try to run in your local machine the test with `make test-all`

**Comment by huuquan1994** [comment]
> well most of the tests are still not passing in this PR. I suggest to try to run in your local machine the test with `make test-all`

I modified the code a bit and as I see it passed all the tests.
Note: I actually have no experience with code testing. So, correct me if I'm wrong!

**Comment by huuquan1994** [comment]
> > well most of the tests are still not passing in this PR. I suggest to try to run in your local machine the test with `make test-all`
> 
> I modified the code a bit and as I see it passed all the tests. Note: I actually have no experience with code testing. So, correct me if I'm wrong!

OK. I'll do the test when I have time.

**Comment by edgarriba** [comment]
Perfect, we'd like to include in the release on Monday. Let me know if you need help, I could take it over the weekend. Do not hesitate to join our slack for direct supply.

**Comment by huuquan1994** [comment]
> Perfect, we'd like to include in the release on Monday. Let me know if you need help, I could take it over the weekend. Do not hesitate to join our slack for direct supply.

Hi, if you want to include this in the release on Monday, could you help me to take a look at the code. I'm not quite sure if I can do it on the weekend.
But if you can wait a bit longer, I think I can handle it.

**Comment by edgarriba** [comment]
@shijianjian please merge when it's in green lghts

**Pull request merged**